<launch>
    <!--
        Includes
    -->
    <include file="$(find fsm)/launch/fsm_node.launch">
        <arg name="veh" value="$(env VEHICLE_NAME)"/>
        <arg name="param_file_name" value="lane_following"/>
    </include>
    
    <!--
        Node ready flags.
    -->
    <group ns="flags">
        <param name="test"                     type="string" value="false"/>
        <param name="wheel_driver_ready"       type="string" value="false"/>
    </group>

    <!--
        Drivers for sensor and actuator components.
    -->
    <group ns="drivers">
        <!--
            Driver for the robot wheels.
        -->
        <group ns="wheels">
            <!--
                Wheel driver topics.
            -->
            <group ns="topics">
                <group ns="inputs">
                    <param name="v_pid_in"     type="string" value="/pid/linear_velocity/out"/>
                    <param name="omega_pid_in" type="string" value="/pid/angular_velocity/out"/>
                </group>
            </group>

            <!--
                Wheel driver nodes.
            -->
            <node pkg="lab4" name="wheel_driver_node" type="wheel_driver_node.py"/>
        </group>
    </group>
</launch>